src/Symfony/Component/Validator/Mapping/Loader/AbstractLoader.php

```diff
@@ -97,9 +97,17 @@ protected function newConstraint(string $name, mixed $options = null): Constrain
                 return new $className($options['value']);
             }
 
+            if (array_is_list($options)) {
+                return new $className($options);
+            }
+
             return new $className(...$options);
         }
 
+        if ($options) {
```

Comment:
if the deprecation is conditional, shouldn't we call the constructor without any argument when there is no option instead of calling it with a falsy argument ?